Releasing process updates (#3911)
grafana/k6: .github/ISSUE_TEMPLATE/hotfix-release.md

---
name: k6 Hotfix release
about: Bugs or security issues have been identified and they requires a patch release.
title: 'k6 hotfix release v<major>.<minor>.<patch>'
labels: ["release"]
---

**Release Date**:

25th May 2024 <<  TODO: WRITE HERE THE UPDATED RELEASE DATE

## Release Activities

> Note, the assumption is a minor branch release already exists from generated from the latest minor release. It has the name in the form `release-v<major>.<minor>.x`.

Reviewer: ```suggestion
Note, the assumption is a minor branch release already exists from the latest minor release. This branch is named in the format `release-v<major>.<minor>.x`.
```

#### Documentation

> - [ ] If required, open and merge pull-request from `main` applying the changes to the affected version.

Reviewer: Strictly speaking, that's outside the scope of this PR but... what's the plan for [docs](https://github.com/grafana/k6-docs/tree/main/docs/sources)? 
Have a folder per minor? Like: `v1.0.x`,  `v1.1.x`, etc?

Reviewer 2: Good question 👀 @heitortsergent do you maybe have insights on how you folks handle that in other projects?

Reviewer 3: I don't have insights into how other projects handle minor versions, and we also handle versions differently than them.

@jdbaldry would you have any insight here? Is it just a matter of renaming/creating folders for minor versions?

Reviewer 4: I'm not sure I understand, sorry. Are you not already maintaining a directory of documentation per minor version of k6?

Reviewer 3: Oh yes, I misread it too, I'm sorry. I thought this was in case we had to release a patch version.

For other projects, they would just have a branch for each minor version, which is the equivalent of our folders in k6-docs. Is that right @jdbaldry?

Reviewer 4: Yeah, most repositories have long-lived version branches for future patching and such, where the docs are maintained alongside the code. Since grafana/k6-docs is a docs-only repository, it's simpler to just have directories in the main branch instead.

> - [ ] If required, open and merge pull-request from `main` applying the changes to the affected version.

Reviewer 2: ```suggestion
- [ ] If required, open and merge pull-request from `master` applying the changes to the affected version.
```

#### In k6 repository, if the defect affects the latest release and the `master` branch

> - [ ] Open, get approvals and merge on `master` the patch for the defect.

Reviewer 5: I believe there is a "gray zone" when the defect affects the master, but `master` contains the changes in the affected logic.

Author: In that case, we need to patch each version independently. `master` gets a version, and the affected branches another dedicated version.
I don't see a better solution for that case. :thinking:

Author: Integrated

- [ ] Open, get approvals and merge on `master` the release notes. Ensure that all merged pull-requests are referenced.
- [ ] Switch on the release branch and cherry-pick the following changes:
      1. the patch for the defect

>       2. the release notes

Reviewer: TODO: Explore the backport GitHub Action from https://github.com/grafana/grafana-github-actions, any idea on where to add this as an item to work on in the future? 🤔

Author: I guess the best way is to open an issue for it. Can you take care of it, please?

- [ ] Open, get approvals and merge on the already existing release branch a separate pull-request for bumping [the k6 Go project's version](https://github.com/grafana/k6/blob/master/lib/consts/consts.go#L11-L12). Note, `master` should be already on a next major/minor version.
- [ ] Pull locally the previously merged changes.

> - [ ] Create and push a new tag of the form `vX.Y.Z` using git: `git tag v0.5x.0 -m "v0.5x.0"`.

Reviewer 2: Just to make sure I don't misunderstand, the `x` in the concrete git tag command is supposed to be a concrete number, not an actual `x` character correct?

Reviewer: I understand it like that, yeah, but let's wait for @codebien's confirmation.

Author: @oleiade Yes, it is correct. What is creating the doubt? How can we improve it?
